### PR TITLE
Refactoring of `CryptoTransferTransitionLogic` 

### PR DESCRIPTION
**Pull request by joan41868**
**Description**:

This PR modifies `CryptoTransferTransitionLogic` to match the other already refactored operations.
Changes made are small, as they are a part of an **incremental** refactoring, and they will be beneficial for the ongoing refactoring of the transfers. 

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/src/test/java/com/hedera/services/txns/crypto/CryptoTransferTransitionLogicTest.java [hunk 1]
```diff
@@ -210,11 +203,10 @@ void shortCircuitsToImpliedTransfersValidityIfNotAvailableInSpan() {
 		given(impliedTransfersMarshal.unmarshalFromGrpc(cryptoTransferTxn.getCryptoTransfer()))
 				.willReturn(impliedTransfers);
 
-		// when:
-		subject.doStateTransition();
+		// when & then:
+		assertThrows(InvalidTransactionException.class, () -> subject.doStateTransition());
 
 		// then:
-		verify(txnCtx).setStatus(TRANSFERS_NOT_ZERO_SUM_FOR_TOKEN);
```
Comment (qnswirlds) [inline hedera-node/src/test/java/com/hedera/services/txns/crypto/CryptoTransferTransitionLogicTest.java:217]: same here for this `assertThrows`

File: hedera-node/src/test/java/com/hedera/services/txns/crypto/CryptoTransferTransitionLogicTest.java [hunk 2]
```diff
@@ -262,10 +254,7 @@ void translatesUnknownException() {
 		givenValidTxnCtx(withAdjustments(a, -2L, b, 1L, c, 1L));
 
 		// when:
-		subject.doStateTransition();
-
-		// then:
-		verify(txnCtx).setStatus(FAIL_INVALID);
+		assertThrows(Exception.class, () -> subject.doStateTransition());
```
Comment (qnswirlds) [inline hedera-node/src/test/java/com/hedera/services/txns/crypto/CryptoTransferTransitionLogicTest.java:257]: and here

**Review by qnswirlds — CHANGES_REQUESTED** [review]
Please see my inline comments. Overall LGTM!

File: hedera-node/src/test/java/com/hedera/services/txns/crypto/CryptoTransferTransitionLogicTest.java (hunk 2 above)
Comment (joan41868) [reply hedera-node/src/test/java/com/hedera/services/txns/crypto/CryptoTransferTransitionLogicTest.java]: Regarding this - I've checked it carefully, and we no longer need this test. Reasons:
* `public boolean tryTransition(@NotNull TxnAccessor accessor)` - this is the signature the function, which gives us the accessor and the test gives us a null `TxnAccessor`
* Exceptions are handled with validation in the refactored transitions.

File: hedera-node/src/test/java/com/hedera/services/txns/crypto/CryptoTransferTransitionLogicTest.java [hunk 3]
```diff
@@ -257,17 +255,6 @@ void computesFailureIfImpliedTransfersNotInSpan() {
...
-	void translatesUnknownException() {
-		givenValidTxnCtx(withAdjustments(a, -2L, b, 1L, c, 1L));
-
-		// when:
-		subject.doStateTransition();
-
-		// then:
-		verify(txnCtx).setStatus(FAIL_INVALID);
-	}
-
```
Comment (qnswirlds) [inline hedera-node/src/test/java/com/hedera/services/txns/crypto/CryptoTransferTransitionLogicTest.java:270]: This test was not well set up. I am ok with removing it now. We will add another test to cover the `FAIL_INVALID` case.

**Review by qnswirlds — COMMENTED** [review]
Looks much better! Thanks @joan41868! I still have a few concerns as in my inline comments. Thanks!

**Review by qnswirlds — APPROVED** [review]
@joan41868, we can merge this PR now. I will take care of those minor concerns that I mentioned later in another PR. Thank you for your work!

**Pull request merged**